frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-form/upload-pvc-form.tsx

```diff
@@ -0,0 +1,375 @@
+          let radioObj = null;
+          const disabled = !allowedAccessModes.includes(radio.value);
+
+          allowedAccessModes.forEach((mode) => {
+            const checked = !disabled ? radio.value === accessMode : radio.value === mode;
+            radioObj = (
+              <RadioInput
+                {...radio}
+                key={radio.value}
+                onChange={handleAccessMode}
+                inline
+                disabled={disabled}
+                checked={checked}
+                aria-describedby="access-mode-help"
+                name="accessMode"
+              />
+            );
+          });
+
+          return radioObj;
+        })}
+        <p className="help-block" id="access-mode-help">
+          {accessModeHelp}
+        </p>
+      </div>
+      <label className="control-label co-required" htmlFor="request-size-input">
+        Size
+      </label>
+      <RequestSizeInput
+        name="requestSize"
+        required
+        onChange={handleRequestSizeInputChange}
+        defaultRequestSizeUnit={requestSizeUnit}
+        defaultRequestSizeValue={requestSizeValue}
+        dropdownUnits={dropdownUnits}
+        describedBy="request-size-help"
+        inputID="request-size-input"
+      />
+      <p className="help-block" id="request-size-help">
+        Desired storage capacity.
+      </p>
+    </div>
+  );
+};
+
+export const UploadPVCPage: React.FC<UploadPVCPageProps> = (props) => {
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [fileValue, setFileValue] = React.useState(null);
+  const [fileName, setFileName] = React.useState('');
+  const [error, setError] = React.useState('');
+  const [isAllocating, setIsAllocating] = React.useState(false);
+  const [dvObj, setDvObj] = React.useState(null);
+
+  const { uploads, uploadData } = React.useContext(CDIUploadContext);
+  const { namespace } = props;
+  const title = 'Upload Data to Persistent Volume Claim';
+
+  const save = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();
+    if (!fileName) {
+      setError('File input is missing');
+    } else {
+      setError('');
+      setIsAllocating(true);
+      setIsSubmitting(true);
+      // eslint-disable-next-line promise/catch-or-return
+      createUploadPVC(dvObj).then(
+        ({ token }) => {
+          setIsAllocating(false);
+          uploadData(fileValue, token, getName(dvObj), getNamespace(dvObj));
+        },
+        ({ message }: { message: string }) => {
+          setIsAllocating(false);
+          setError(message || 'Could not create persistent volume claim.');
+        },
+      );
+    }
+  };
+
+  const handleFileChange = (value, filename) => {
+    setFileName(filename);
+    setFileValue(value);
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <div
+        className={cx('co-m-pane__body co-m-pane__form', {
+          'kv--create-upload__hide': isSubmitting,
```

Comment:
same as the other one, just dont render while submitting, no css required

Reply:
https://github.com/openshift/console/pull/5765#discussion_r452801126

Reply:
ok, makes sense :)